### PR TITLE
[FLINK-8778] Migrate queryable state ITCases to use MiniClusterResource

### PR DESCRIPTION
**Pull request by aljoscha**
This sits on top of #5589.

### CODE REVIEW

File: flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java [hunk 1]
```diff
@@ -75,6 +82,10 @@ public MiniClusterResource(
 		this.miniClusterType = Preconditions.checkNotNull(miniClusterType);
 	}
 
+	public ClusterClient getClusterClient() {
```
Comment (zentol) [inline flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java]: return `ClusterClient<?>`?

File: flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java [hunk 2]
```diff
@@ -981,7 +981,7 @@ public Configuration getFlinkConfiguration() {
 	 * @param jobGraph The JobGraph to be submitted
 	 * @return JobSubmissionResult
 	 */
-	protected abstract JobSubmissionResult submitJob(JobGraph jobGraph, ClassLoader classLoader)
+	public abstract JobSubmissionResult submitJob(JobGraph jobGraph, ClassLoader classLoader)
```
Comment (zentol) [inline flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java]: we could deprecate all other submit-like methods that the `RestClusterClient` doesn't override.

Comment (aljoscha) [reply flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java]: Should do, but is that in the scope of this PR?

File: flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java (hunk 1 above)
Comment (aljoscha) [reply flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterResource.java]: fixing

**Comment by aljoscha** [comment]
@zentol I'm fixing things on #5589 because this PR is only about the last commit.

File: flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java (hunk 2 above)
Comment (zentol) [reply flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java]: well...i guess wrong usages will be catched by the tests....

**Comment by aljoscha** [comment]
Closing in favour of #5589.

**Pull request closed without merging**